src/transformers/modeling_utils.py

```diff
@@ -968,7 +968,10 @@ def _load_state_dict_into_meta_model(
             if is_fsdp_enabled() or is_deepspeed_zero3_enabled():
                 module, tensor_name = get_module_from_name(model, param_name)
                 value = getattr(module, tensor_name)
-                value = type(value)(value.data.to("cpu"), **value.__dict__)
+                val_kwargs = {}
+                if hasattr(module, "weight") and module.weight.__class__.__name__ == "Int8Params":
+                    val_kwargs["requires_grad"] = False
```

Comment:
since this is only applicable to `Int8Param` which I suppose is specific to a certain lib, why is this not going in the `create_quantized_param` ? (if it's bnb's int8 then in bnb)

Reply:
`create_quantized_param` already attempts to set requires_grad=False  (https://github.com/huggingface/transformers/blob/main/src/transformers/quantizers/quantizer_bnb_8bit.py#L215), but that attribute isn't part of the `__dict__` when we're trying to move it to cpu for for fsdp/deepspeed

Reply:
mmmm then is there no other way to just preserve the actual tensor (here we are acessing value.data so that's were we lose the info from the create quantize param no?

Reply:
Yeah, I'm assuming we can't simply move the tensor and that's why the current code has to recreate the tensor on meta/cpu and it loses the requires_grad info

Reply:
```suggestion
                 value = torch.tensor(value.data.to("cpu"), **value.__dict__)
```
I am guessing you tried, but something as simple as this doesn't work?

Reply:
I gave that a shot and that doesn't work. `TypeError: tensor() got an unexpected keyword argument 'CB'`